lib/Sema/CSGen.cpp

```diff
@@ -1152,6 +1152,19 @@ namespace {
       // `_ = nil`, let's diagnose it here because solver can't
       // attempt any types for it.
       auto *parentExpr = CS.getParentExpr(expr);
+      if (auto *PE = dyn_cast<ParenExpr>(parentExpr))
```

Comment:
since `PE` is not used, it could be replaced with `parentExpr && isa<ParenExpr>(parentExpr)`